tendrl/alerting/objects/alert/__init__.py

```diff
@@ -0,0 +1,125 @@
+        resource=None,
+        current_value=None,
+        tags=None,
+        alert_type=None,
+        severity=None,
+        significance=None,
+        ackedby=None,
+        acked=None,
+        pid=None,
+        source=None,
+        *args,
+        **kwargs
+    ):
+        super(Alert, self).__init__(*args, **kwargs)
+        self.alert_id = alert_id
+        self.node_id = node_id
+        self.time_stamp = time_stamp
+        self.resource = resource
+        self.current_value = current_value
+        self.tags = tags
+        self.alert_type = alert_type
+        self.severity = severity
+        self.significance = significance
+        self.ackedby = ackedby
+        self.acked = acked
+        self.pid = pid
+        self.source = source
+
+
+class _Alert(EtcdObj):
+    """A table of the node context, lazily updated
+
+    """
+    __name__ = 'alerting/alerts/%s'
+    _tendrl_cls = Alert
+
+    def render(self):
+        self.__name__ = self.__name__ % self.alert_id
+        return super(_Alert, self).render()
+
+
+@to_singleton
+class AlertUtils(object):
+    def __init__(self):
+        self.etcd_client = tendrl_ns.etcd_orm.client
+
+    def equals(self, alert1, alert2):
+        return (
+            alert1.alert_id == alert2.alert_id and
+            alert1.node_id == alert2.node_id and
+            alert1.time_stamp == alert2.time_stamp and
+            alert1.resource == alert2.resource and
+            alert1.current_value == alert2.current_value and
+            alert1.tags == alert2.tags and
+            alert1.alert_type == alert2.alert_type and
+            alert1.severity == alert2.severity and
+            alert1.significance == alert2.significance and
+            alert1.ackedby == alert2.ackedby and
+            alert1.acked == alert2.acked and
+            alert1.pid == alert2.pid and
+            alert1.source == alert2.source
+        )
+
+    def update(self, alert1, existing_alert):
+        time_stamp = existing_alert.time_stamp
+        if (
+            alert_severity_map[alert1.severity] < alert_severity_map[
+                existing_alert.severity] and
+            alert_severity_map[alert1.severity] == alert_severity_map[
+                'INFO']
+        ):
+            time_stamp = alert1.time_stamp
+            alert1.ackedby = 'Tendrl'
+            alert1.acked = True
+        alert1.alert_id = existing_alert.alert_id
+        alert1.time_stamp = time_stamp
+        return alert1
+
+    def is_same(self, alert1, alert2):
+        if alert1.resource != alert2.resource:
+            return False
+        if 'plugin_instance' in alert1.tags:
+            if 'plugin_instance' not in alert2.tags:
+                return False
+            else:
+                if (
+                    alert1.tags['plugin_instance'] !=
+                    alert2.tags['plugin_instance']
+                ):
+                    return False
+        if 'Tendrl_context.cluster_id' in alert1.tags:
```

Comment:
Now its called `TendrlContext.integration_id`. You might need related changes in object definition as well.